### ISSUE TITLE
Not compileing main source when compile unit test sources

### ISSUE THREAD

**Issue by nenick** [nenick/AndroidStudioAndRobolectric#7]
When you try to run unit tests from a library module then you must compile the main sources first. `gradle lib:assembleDebug`

Issue is tracked at https://code.google.com/p/android/issues/detail?id=158251

**Comment by vanniktech** [comment]
Awesome. Works perfectly fine for me. If you want to just use `./gradlew clean test` you have to add this to build.gradle:

``` groovy
tasks.withType(Test) {
    dependsOn assembleDebug
}
```

Although than you may get a problem with the `testRelease` task. Therefore execute:

`./gradlew clean test -x testRelease`

**Comment by nenick** [comment]
here is an enhanced version

```
afterEvaluate { project ->
    android.libraryVariants.each { variant ->
        tasks.getByName("test${ variant.name.capitalize()}") {
            dependsOn "assemble${ variant.name.capitalize()}"
        }
    }
}
```

**Comment by jaredsburrows** [comment]
Isnt this issue a duplicate of https://github.com/nenick/AndroidStudioAndRobolectric/issues/5?

**Comment by nenick** [comment]
They are two different issues. 

This one miss the compiled main classes of the current library module under test. Google issue was merged into https://code.google.com/p/android/issues/detail?id=156027

The other one miss resource files specified under test (and is already fixed with 1.2beta).

**Comment by nenick** [comment]
Fixed with gradle android plugin 1.2.+
